### PR TITLE
GeckoCIRCUITS collection

### PR DESCRIPTION
**Pull request by tinix84**
Hi I did a collection of many material I found over web, and try to organize my best. The idea was to keep the src as you did and build around example, etc.. so that the other people can start to play with it.
Next steps for me is to improve the docs, build a github website and a python designer for power electronic on top of gecko..
I hope that is fine for you

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -11,6 +14,8 @@ All the source is from the original author. We just modified it such that it run
 
 Technokrat GmbH takes no credit whatsoever on the original codebase.
 
+ciao si il corso e molto interessante
```
Comment (Yatekii) [inline README.md]: I think this got forgotten :) Please remove this :)

**Review by Yatekii — CHANGES_REQUESTED** [review]
Hi Riccardo
Thanks for the contribution and the work you put in :)

Pretty much everything looks fine to me!

Two things I would like to see changed:
- Could you please be so kind and remove the built binaries. Binaries should never be checked into source versioning.
This codebase is already in terrible shape and I would like to keep it from drifting into oblivion ;)
If you would like to add the binaries for users, the https://github.com/technokrat/gecko/releases section in Github would be the right place to do :)
I'd happily add you as a collaborator, so you can do so! 
- I added a small note about a comment in Italian that might have gone missing. Even tho I manage to understand some Italian, most users wont, so I think the right course of action is to keep all the code in English (yes, I know, parts are in German, which is not okay either. I just didn't find the time to clean the entire codebase).

All the .ipes files are quite a mess, but I think we can correct that in the future.

Please be aware that this was mostly just a clean build we can use at https://technokrat.ch, also going into the future.
I originally planned to rewrite components in Rust and add a model for a noise shaper. But the code base is in such terrible shape that it takes any motivation to do so :(
I'll happily maintain this repo and give my feedback on PRs, but I most likely wont find time & motivation to add any new code, especially since I don't like Java and because of the codebase.
Just to give you a heads up, but you might know from the lack of activity on this repo already ;)

What do you plan to add in the form of Python tools? Bindings to the Gecko API? I would love that! More often than not it would be really nice to be able to interact with Gecko from within Python for example for simulating a noise shaper ;)

Best,
Noah

**Comment by tinix84** [comment]
cool thanks for the suggestions! 
- I was not aware of the release section (i'm a stupid HW engineer), what is the best way to do that? zip file of the 2 folder jdk8 and jdk13?
- Sorry regarding the ipes mess but my idea was to preserve the original folder structures, to not do to many change in one commit, in compare the actual status (I have ask Mr. Musing support but seem to be absent in the last period)
- Regarding python stuff with gecko, I based my work on https://github.com/MauererM/pygeckocircuits, but the API interface is not working with jdk13 and I dont know why (win10??)

File: README.md (hunk 1 above)
Comment (tinix84) [reply README.md]: done

**Comment by tinix84** [comment]
done everything requested

**Comment by Yatekii** [comment]
Looking great, thanks!

- No worries, I wasn't a daily Github frequenter until like 1.5 years ago ;) I'm a stupid SW engineer, so we are united in stupidity ;P
Zip with the two folders should suffice :)

- Oh, not at all, I wasn't blaming the mess on you! I would just love to see this cleaned up in the future, but I didn't find time to put in myself yet, so we'll see how it develops :) For now it is certainly ok!

- Ah nice, I see, you know all the folks from PES, I know too ;) I could have a look sometime I guess. I could also ask Mario if he had any luck with it. Maybe the bindings changed between versions.

Is there a reason you closed-reopened-reclosed this issue? I am assuming this was by mistake and will merge this :)

Thank you for the work!

**Comment by tinix84** [comment]
Hi 
I dont know personally PES people, just some of them because of Brusa and NTB (ex NTB now is OST :))
regarding the python designer and interface I would like to have some suggestion, would you have some time to review it?
I have written pyplecs package https://github.com/tinix84/pyplecs  to automatize plecs and I would like to do similar stuff for gecko. My main problems are
1. gecko is just listening on one port -> no possible parallel simulation (plecs is even worse but I have automatize the GUI to do a workaround)
2. my wish would be to use the abstraction to cover with similar class gecko and plecs, such that the external interface are the same
3. I would like to use metrics and code review on Github, do you know how? I have implemented test unit but I'm not sure are correct

At the end the idea is to use AI to design power electronic, helping the design to choose optimal design approach

**Pull request merged**